### PR TITLE
refactor: (controller) error handling

### PR DESCRIPTION
**Pull request by Roxxas96**
I re implemented error handling through all interfaces, now we have a more readable nested error handling

### CODE REVIEW

**Comment by sameo** [comment]
@Roxxas96 Could you please do the format fixes for this PR, so that we can have this one and the kudoctl PRs merged?

**Comment by Roxxas96** [comment]
Hey ! This PR is not ready to be merged yet, their are some bad practices that i didn't overcome yet. Unfortunately I am in vacation so I probably won't be able to finish this one until next week. Is it really urgent ? If it is I can rebase, merge and continue later.

**Comment by lucido-simon** [comment]
Hi, to add to Samuel's comment, this is blocking the kudoctl PR. Maybe you could quickly define a task list, and I can take over the PR during your vacation?

**Comment by Roxxas96** [comment]
If it's just a matter of API response format i can handle it. We just need to synchronize ourselves on the structure. I sent you some questions on your PR @lucido-simon

**Comment by Roxxas96** [comment]
What have been done :

- add generic API response to build a standard for all our responses
- add better described errors on every servies, controllers & interfaces
- add better logging for more observability

What still need to be done (in another PR) :

- add more observability over controllers & services (which have not been reviewed)
- fix the verbosity problem during node status update
- `namespace` route has not been reviewed yet
- Too many wrong error handling such as `unwrap()` or `match` madness

**Comment by Roxxas96** [comment]
@lucido-simon The PR is ready to be merged if you want. You said that you needed it fast.

**Comment by lucido-simon** [comment]
> @lucido-simon The PR is ready to be merged if you want. You said that you needed it fast.

Yeah sorry, was quite busy and didn't have time to review the PR.

File: controller/lib/src/external_api/generic/model.rs [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+    pub data: Option<String>,
+    pub metadata: APIResponseMetadata,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct APIResponseMetadata {
+    pub error: Option<String>,
+    pub message: Option<String>,
+    pub count: Option<u32>,
+    pub status: u16,
```
Comment (lucido-simon) [inline controller/lib/src/external_api/generic/model.rs]: I'm not sure why you would add a status here ? The status is in the HTTP headers already, no need to duplicate the information here.

Comment (Roxxas96) [reply controller/lib/src/external_api/generic/model.rs]: I was just adding this to metadata so this object can have all information needed (as well as the status) but I can remove it in fact, it won't be so useful for us.

**Review by iverly — APPROVED** [review]

**Comment by iverly** [comment]
@sameo is it possible to give the write access to the repository to some of us ? (we have chosen the ones who will maintain this repository) thanks :)

**Comment by sameo** [comment]
> @sameo is it possible to give the write access to the repository to some of us ? (we have chosen the ones who will maintain this repository) thanks :)

Done.

**Pull request merged**